docs/reference/configuration/configuration.md

```diff
@@ -74,15 +87,15 @@ When compiling or exporting, the configuration system generates C preprocessor m
 <file truncated for brevity>
 ```
 
-The name of the macro for a configuration parameter is either a prefixed name or explicitly specified by `macro_name`. The configuration system constructs a prefixed name from the prefix `MBED_CONF_`, followed by the name of the library or `APP`, followed by the name of the parameter. The configuration system then capitalizes the prefixed name and converts it to a valid C macro name. For example, the configuration system converts the `random_max_start_delay` configuration parameter in the library `cellular` to `MBED_CONF_CELLULAR_RANDOM_MAX_START_DELAY`.
+The name of the macro for a configuration parameter is either a prefixed name or explicitly specified by `macro_name`. The configuration system constructs a prefixed name from the prefix `MBED_CONF_`, followed by the name of the library or `APP`, followed by the name of the parameter. The configuration system then capitalizes the prefixed name and converts it to a valid C macro name. For example, the configuration system converts the `random_max_start_delay` configuration parameter in the library `cellular` to `MBED_CONF_CELLULAR_RANDOM_MAX_START_DELAY`. As for using `macro_name` field, it is strongly encouraged to avoid its use unless the intent is to support an already defined macro that is heavily used in Mbed OS, or user applications.
```

Comment:
It would be good to add a sentence mentioning that "macro_name" usage is not recommended because its easy to figure out that MBED_CONF_ flags are driven by config files.

Reply:
Apologies if he added this after your comment, but he mentions:
>As for using `macro_name` field, it is strongly encouraged to avoid its use unless the intent is to support an already defined macro that is heavily used in Mbed OS, or user applications.

Reply:
@kegilbert - Yes he mentions that, what I meant was it might be useful to say why we are not discouraging people from using 'macro_name' instead of just saying we discourage that usage. Not a big deal, just a thought.

Reply:
I can add little more detail to it.